Sources/Sentry/SentryScope.m

```diff
@@ -194,7 +206,18 @@ - (void)clearBreadcrumbs
 - (NSArray<SentryBreadcrumb *> *)breadcrumbs
 {
     @synchronized(_breadcrumbArray) {
-        return _breadcrumbArray.copy;
+        NSMutableArray<SentryBreadcrumb *> *crumbs = [NSMutableArray new];
+        for (int i = 0; i < _maxBreadcrumbs; i++) {
+            // Crumbs use a ring buffer. We need to start at the current crumb to get the
+            // crumbs in the correct order.
+            NSInteger index = (_currentBreadcrumb + i) % _maxBreadcrumbs;
+
+            if (index < _breadcrumbArray.count) {
+                [crumbs addObject:_breadcrumbArray[index]];
+            }
```

Comment:
Ups, no, that doesn't work because we always start the loop at 0. When the ring buffer isn't full yet, and the `_currentBreadcrumbIndex` is for example 70, we need to continue until we reach % _maxBreadcrumbs reaches 0. So the first 30 attempts of retrieving something from the array fail, but then we succeed.

Reply:
I dont get  how can `_currentBreadcrumbIndex` start at 70, we're always reseting it to 0 when the `_breadcrumbArray` is clear. 
Also `_currentBreadcrumbIndex` can never be bigger than `_breadcrumbArray.count` + 1 otherwise it crashes if we try to set a breadcrumbs in that index.